src/content/docs/fr/core-concepts/astro-components.mdx

```diff
@@ -1,9 +1,305 @@
 ---
 title: Composants
-description: Une introduction à la syntaxe des composants en.astro.
+description: Une introduction à la syntaxe des composants en .astro.
 i18nReady: true
 ---
 
+**Les composants Astro** sont les blocs de base de tout projet Astro. Ce sont des composants de modélisation HTML uniquement, sans exécution côté client. Vous pouvez repérer un composant Astro par son extension de fichier : `.astro`.
+
+Les composants Astro sont extrêment flexibles. Souvent, un composant Astro contiendra une **interface utilisateur réutilisable sur la page**, comme un en-tête ou une carte de profil. À d'autres moments, un composant Astro peut contenir un petit extrait de HTML, comme une collection de balises `<meta>` courantes qui facilitent le travail avec le référencement. Les composants Astro peuvent même contenir une mise en page de page entière.
+
+La chose la plus importante à savoir sur les composants Astro est qu'ils **ne rendent pas sur le client**. Ils rendent en HTML soit au moment de la construction ou à la demande en utilisant le [rendu côté serveur (SSR)](/fr/guides/server-side-rendering/). Vous pouvez inclure du code JavaScript dans votre frontmatter de composant, et tout sera supprimé de la page finale envoyée aux navigateurs de vos utilisateurs. Le résultat est un site plus rapide, avec une empreinte JavaScript nulle ajoutée par défaut.
+
+Quand votre composant Astro a besoin d'interactivité côté client, vous pouvez ajouter [des balises HTML `<script>` standard](/fr/guides/client-side-scripts/) ou [composants de framework d'interface utilisateur réutilisable](/fr/core-concepts/framework-components/#hydratation-des-composants-interactifs).
+
+
+## Structure du composant
+
+Un composant Astro est composé de deux parties principales : le **Script du composant** et le **Template du composant**. Chaque de ces parties s'occupe de faire une tâche différente, mais ensemble ils fournissent un cadre qui es à la fois facile à utiliser et assez expressif pour gérer tout ce que vous pourriez vouloir construire.
+
+```astro title="src/components/EmptyComponent.astro"
+---
+// Script du composant (JavaScript)
+---
+<!-- Template du composant (HTML + Expressions JS) -->
+```
+
+### Le Script du composant
+
+Astro utilise des barres de code (`---`) pour identifier le script du composant dans votre composant Astro. Si vous avez déjà écrit du Markdown avant, vour pourriez être déjà familier avec un concept similaire appelé *frontmatter.* L'idée d'un script de composant d'Astro a été directement inspirée par ce concept.
+
+Vous pouvez utiliser le script du composant pour écrire du code JavaScript qui vous aidera à construire votre template. Cela peut inclure :
```

Comment:
I believe this should be a non-breaking space: `&nbsp;:` (cf [Espacement avant et après les signes de ponctuation et les symboles](https://vitrinelinguistique.oqlf.gouv.qc.ca/22039/la-typographie/espacement/espacement-avant-et-apres-les-signes-de-ponctuation-et-les-symboles))